### ISSUE TITLE
Aliasing records pattern matching break if space exists

### ISSUE THREAD

**Issue by Yanpas** [ndmitchell/hlint#486]
This snippet is parsed successfully:
```
data X = X {unX :: Int, unY :: Bool}
foo :: X -> Bool
foo x@X{unY = bb} = True
```

But after adding spaces:
```
data X = X {unX :: Int, unY :: Bool}
foo :: X -> Bool
foo x @ X{unY = bb} = True
```
I get '[hlint] Parse error'

I'm using haskell-ide-engine in vs code
Version: `PS D:\dev\pandoc> hlint --version
HLint v2.1, (C) Neil Mitchell 2006-2018`

If I run hlint from command line, I don't see this warning...
```
hlint .\src\Text\Pandoc\Readers\Man.hs
.\src\Text\Pandoc\Readers\Man.hs:47:1: Suggestion: Unnecessary hiding
Found:
  ?[32;4mimport?[0m?[0m ?[0m?[0mText?[0m?[36m.?[0m?[0mPandoc?[0m?[36m.?[0m?[0mParsing?[0m?[0m ?[0m?[0mhiding?[0m?[0m ?[0m?[36m(?[0m?[0menclosed?[0m?[36m)?[0m
Why not:
  ?[32;4mimport?[0m?[0m ?[0m?[0mText?[0m?[36m.?[0m?[0mPandoc?[0m?[36m.?[0m?[0mParsing?[0m

.\src\Text\Pandoc\Readers\Man.hs:75:49: Warning: Redundant bracket
Found:
  ?[36m(?[0m?[0mManState?[0m?[36m{?[0m?[0mpState?[0m?[0m ?[0m?[31m=?[0m?[0m ?[0m?[0mdef?[0m?[36m,?[0m?[0m ?[0m?[0mrState?[0m?[0m ?[0m?[31m=?[0m?[0m ?[0m?[0mdef?[0m?[36m}?[0m?[36m)?[0m
Why not:
  ?[0mManState?[0m?[36m{?[0m?[0mpState?[0m?[0m ?[0m?[31m=?[0m?[0m ?[0m?[0mdef?[0m?[36m,?[0m?[0m ?[0m?[0mrState?[0m?[0m ?[0m?[31m=?[0m?[0m ?[0m?[0mdef?[0m?[36m}?[0m

.\src\Text\Pandoc\Readers\Man.hs:79:13: Suggestion: Move brackets to avoid $
Found:
  ?[35m"Pandoc: "?[0m?[0m ?[0m?[36m++?[0m?[0m ?[0m?[36m(?[0m?[0mshow?[0m?[0m ?[0m?[36m$?[0m?[0m ?[0m?[0munMeta?[0m?[0m ?[0m?[0mm?[0m?[36m)?[0m
Why not:
  ?[35m"Pandoc: "?[0m?[0m ?[0m?[36m++?[0m?[0m ?[0m?[0mshow?[0m?[0m ?[0m?[36m(?[0m?[0munMeta?[0m?[0m ?[0m?[0mm?[0m?[36m)?[0m

.\src\Text\Pandoc\Readers\Man.hs:115:20: Suggestion: Use unwords
Found:
  ?[34mconcat?[0m?[0m ?[0m?[36m$?[0m?[0m ?[0m?[0mintersperse?[0m?[0m ?[0m?[35m" "?[0m?[0m ?[0m?[0margs?[0m
Why not:
  ?[34munwords?[0m?[0m ?[0m?[0margs?[0m

.\src\Text\Pandoc\Readers\Man.hs:146:29: Suggestion: Use void
Found:
  ?[34mmodifyRoffState?[0m?[0m ?[0m?[36m(?[0m?[31m\?[0m?[0m ?[0m?[0mrst?[0m?[0m ?[0m?[31m->?[0m?[0m ?[0m?[0mrst?[0m?[36m{?[0m?[0minCodeBlock?[0m?[0m ?[0m?[31m=?[0m?[0m ?[0m?[0minsideCB?[0m?[36m}?[0m?[36m)?[0m?[0m ?[0m?[36m>>?[0m?[0m ?[0m?[0mreturn?[0m?[0m ?[0m?[0m()?[0m
Why not:
  ?[0mControl?[0m?[36m.?[0m?[0mMonad?[0m?[36m.?[0m?[0mvoid?[0m?[0m
  ?[0m?[0m  ?[0m?[36m(?[0m?[0mmodifyRoffState?[0m?[0m ?[0m?[36m(?[0m?[31m\?[0m?[0m ?[0m?[0mrst?[0m?[0m ?[0m?[31m->?[0m?[0m ?[0m?[0mrst?[0m?[36m{?[0m?[0minCodeBlock?[0m?[0m ?[0m?[31m=?[0m?[0m ?[0m?[0minsideCB?[0m?[36m}?[0m?[36m)?[0m?[36m)?[0m

.\src\Text\Pandoc\Readers\Man.hs:180:28: Suggestion: Redundant bracket
Found:
  ?[0mString?[0m?[0m ?[0m?[31m->?[0m?[0m ?[0m?[36m(?[0m?[0mMaybe?[0m?[0m ?[0m?[0mInline?[0m?[36m)?[0m
Why not:
  ?[0mString?[0m?[0m ?[0m?[31m->?[0m?[0m ?[0m?[0mMaybe?[0m?[0m ?[0m?[0mInline?[0m

6 hints
```
But I see unix-terminal special ascii charcters, which are not supported by powershell

**Comment by ndmitchell** [comment]
Thanks for the report. To address the issues separately:

1. Works fine for me and for you on the command line. Sounds like a bug for the haskell-ide-engine.
2. Powershell on Windows 10? I just tried it and it doesn't print out any escape characters. HLint auto detects, and on Windows it usually detects colors are not supported.

**Comment by Yanpas** [comment]
> Sounds like a bug for the haskell-ide-engine.
I'll open an issue there, at least HIE shouldn't confuse users with the source of an error

> Powershell on Windows 10
I've also noticed that `stack` prints some ascii symbols too. Seems like they check color support in the same fashion. Do you know how? Maybe I have some env variable that confuses them. I have WSL installed on my windows, so I'm able to run linux apps on my machine, maybe it's related

**Comment by ndmitchell** [comment]
Maybe the WSL is related, I don't have it on mine. FWIW, Stack prints garbage normally for me, so that's a different bug.

**Comment by Yanpas** [comment]
From https://github.com/haskell/haskell-ide-engine/issues/555

alanz wrote:
> I think the @ symbol is sensitive to preceding space now in GHC, when the VisibleTypeApplication extension is enabled. So possibly hlint is enabling that extension by default.

**Comment by ndmitchell** [comment]
I do indeed turn on type application, but it still works for me even when I do so... I guess the question is how is HLint being invoked by HIE? What about that invocation causes it to fail? Do you have extensions enabled in your `.cabal` file?
